ChangeLog

```diff
@@ -10,6 +10,9 @@ Bugfix
    * Add ecc extensions only if an ecc based ciphersuite is used.
      This improves compliance to RFC 4492, and as a result, solves
      interoperability issues with BouncyCastle. Raised by milenamil in #1157.
+   * Fix overly strict bounds check in ssl_parse_certificate_request()
+     which could lead to valid CertificateRequest messages being rejected.
```

Comment:
I know I'm picky on ChangeLog entries, but I think we should be more explicit about the consequences (handshake failure) and the circumstances (TLS 1.1 and earlier, client-side bug, etc). Also, we should start with the user-oriented description, and put the technical details if any only later. I'd like to suggest something more like

> Fix a bug that caused SSL/TLS clients to incorrectly abort the handshake with TLS versions 1.1 and earlier when the server requested authentication without providing a list of CAs. This was due to an overly strict bounds check in parsing the CertificateRequest message, introduced in Mbed TLS 2.12.0.

Reply:
@mpg That's much clearer, thanks you - I will follow your suggestion.